tests/testAutopas/RegionParticleIteratorTest.cpp

```diff
@@ -65,22 +103,66 @@ TEST_F(RegionParticleIteratorTest, testLinkedCellsRegionParticleIteratorBehavior
   for (auto iterator = lcContainer.getRegionIterator(testRegionMin, _regionMax, autopas::IteratorBehavior::haloOnly);
        iterator.isValid(); ++iterator) {
     iterator->touch();
+    EXPECT_TRUE(utils::inBox(iterator->getR(), testRegionMin, _regionMax)
+                    ? (utils::inBox(iterator->getR(), _boxMin, _boxMax) ? 0 : 1)
+                    : 0)
+        << " particle at [" << iterator->getR()[0] << ", " << iterator->getR()[1] << ", " << iterator->getR()[2] << "]"
+        << " in thread: " << autopas_get_thread_num() << std::endl;
   }
 
   // check the touch using the normal iterator
   for (auto iterator = lcContainer.begin(); iterator.isValid(); ++iterator) {
-    //  std::cout << "id: " << iterator->getID() << " at [" <<
-    //  iterator->getR()[0]
-    //         << ", " << iterator->getR()[1] << ", " << iterator->getR()[2]
-    //              << "] touched:" << iterator->getNumTouched() << std::endl;
-
+    // this is a test for halo only! so we first check whether it's within our region of interest and then whether it's
+    // not in the halo
     EXPECT_EQ(utils::inBox(iterator->getR(), testRegionMin, _regionMax)
```

Comment:
Create a variable for this condition for readability (especially of gtest output on case of fail)

Reply:
https://github.com/AutoPas/AutoPas/pull/143/commits/9c1a93fa659df430846823bb885d3291ad059715